Source/WebCore/platform/audio/cocoa/AudioFileReaderCocoa.cpp

```diff
@@ -219,8 +217,9 @@ std::unique_ptr<AudioFileReaderWebMData> AudioFileReader::demuxWebMData(const ui
         track->setDiscardPadding(discardPadding);
     });
     SourceBufferParser::Segment segment(Ref { buffer.get() });
-    parser->appendData(WTFMove(segment));
-    if (!track || error)
+    auto promise = parser->appendData(WTFMove(segment));
+    ASSERT(promise->isSettled());
+    if (!track || !promise->isResolved())
```

Comment:
Why a promise if it should be resolved synchronously? The previous code seems better.

Reply:
The SourceBufferParser has two mode of operations. One synchronous like here. And one as used by the SourceBuffer where you define the thread on which the callbacks will be called. That is asynchronous. 

We should make the webaudio decoder fully asynchronous too. But it was out of scope for this change.

Reply:
I've made SourceBufferParser::appendData return an `Expected<void, int>` instead so that it can be used synchronously. Will migrate in a follow-up change